src/main/java/com/chessmates/service/EntityServiceImpl.groovy

```diff
@@ -2,130 +2,35 @@ package com.chessmates.service
 import com.chessmates.model.Player
-import com.chessmates.utility.GetPageFunction
-import com.chessmates.utility.LichessApi
-import com.chessmates.utility.LichessResultSet
-import org.apache.commons.lang3.tuple.ImmutablePair
 
 import org.springframework.beans.factory.annotation.Autowired
 import org.springframework.stereotype.Service
 
-import java.util.stream.Collectors
-
 /**
  * Service handling data requests relating to Lichess entities.
  */
 @Service
 class EntityServiceImpl implements EntityService {
 
-    static final String TEAM_NAME = 'scott-logic'
-
-    private LichessApi lichessApi
-    // TODO: Temporary - this wants to come from the database.
-    private Map<ImmutablePair, Game> playerIdToLatestGame = new HashMap<>() // A mapping of player IDs to their latest process game.
+    private LichessDataService lichessDataService
 
     @Autowired
-    EntityServiceImpl(LichessApi lichessApi) {
-        this.lichessApi = lichessApi
+    EntityServiceImpl(LichessDataService lichessDataService) {
+        this.lichessDataService = lichessDataService
     }
 
     /**
-     * All SL players.
+     * Get all Lichess players.
      */
     @Override
-    List<Player> getPlayers() {
-        def fetchPlayerPage = { String teamId, int pageNum -> lichessApi.getPlayers(teamId, pageNum) }
-
-        def neverStop = { t -> false}
-
-        def resultSet = new LichessResultSet<Player>(
-                (GetPageFunction)fetchPlayerPage.curry(TEAM_NAME),
-                neverStop
-        )
-
-        resultSet.get()
-    }
+    // TODO: Replace with fetch from DB.
+    List<Player> getPlayers() { lichessDataService.getPlayers(null) }
 
     /**
      * Get all games between scott logic players.
      */
     @Override
-    List<Game> getAllTeamGames() {
-        getGames(getPlayers())
-    }
-
-    /**
-     * Get new games between scott logic players.
-     */
-    List<Game> getGamesUntilLatestFetched() { getGames(getPlayers(), playerIdToLatestGame) }
-
-    /**
-     * Get only new games for each individual player combination.
-     *
-     * If no map is provided all historical games will be fetched. Go get a coffee!
-     *
-     * @param players The players to get new games for.
-     * @param latestGameMap A map containing the latest game for each pair of opponents.
-     * @return A list of all new games.
-     */
-    private List<Game> getGames(List<Player> players, Map<ImmutablePair, Game> latestGameMap = new HashMap<>()) {
-
-        def fetchGamePage = { Player player, Player opponent, int pageNum -> lichessApi.getGames(player.id, opponent.id, pageNum) }
-
-        def gameIsLatest = { Player player, Player opponent, Game thisGame ->
-            thisGame == latestGameMap.get(new ImmutablePair(player, opponent))
-        }
-
-        def playerCombinations = uniqueCombinations(players)
+    // TODO: Replace with fetch from DB.
+    List<Game> getGames() { lichessDataService.getGames(getPlayers(), null) }
```

Comment:
Maybe we can make a version that does not take a null, but just takes the players